src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1032,205 +1037,181 @@ void Ekf2::run()
 		const bool updated = _ekf.update();
 		perf_end(_perf_ekf_update);
 
-		if (updated) {
-
-			// integrate time to monitor time slippage
-			if (_start_time_us == 0) {
-				_start_time_us = now;
-				_last_time_slip_us = 0;
-
-			} else if (_start_time_us > 0) {
-				_integrated_time_us += sensors.gyro_integral_dt;
-				_last_time_slip_us = (now - _start_time_us) - _integrated_time_us;
-			}
-
-			matrix::Quatf q;
-			_ekf.copy_quaternion(q.data());
-
-			// In-run bias estimates
-			float gyro_bias[3];
-			_ekf.get_gyro_bias(gyro_bias);
-
-			{
-				// generate vehicle attitude quaternion data
-				vehicle_attitude_s att;
-				att.timestamp = now;
-
-				q.copyTo(att.q);
-				_ekf.get_quat_reset(&att.delta_q_reset[0], &att.quat_reset_counter);
+		// integrate time to monitor time slippage
+		if (_start_time_us == 0) {
+			_start_time_us = now;
+			_last_time_slip_us = 0;
 
-				att.rollspeed = sensors.gyro_rad[0] - gyro_bias[0];
-				att.pitchspeed = sensors.gyro_rad[1] - gyro_bias[1];
-				att.yawspeed = sensors.gyro_rad[2] - gyro_bias[2];
+		} else if (_start_time_us > 0) {
+			_integrated_time_us += sensors.gyro_integral_dt;
+			_last_time_slip_us = (now - _start_time_us) - _integrated_time_us;
+		}
 
-				// publish vehicle attitude data
-				if (_att_pub == nullptr) {
-					_att_pub = orb_advertise(ORB_ID(vehicle_attitude), &att);
+		if (updated) {
 
-				} else {
-					orb_publish(ORB_ID(vehicle_attitude), _att_pub, &att);
+			filter_control_status_u control_status;
+			_ekf.get_control_mode(&control_status.value);
+
+			// only publish position after successful alignment
+			if (control_status.flags.tilt_align && control_status.flags.yaw_align) {
```

Comment:
Should only be gated on tilt align. yaw alignment can momentarily go false.

Reply:
Makes sense, I'll change it. Currently it's both in master. https://github.com/PX4/ecl/blob/b26c2d62b80d6bee148fbfe01eb36809bfa721ea/EKF/ekf.cpp#L134